File: api/Data/Models/Customer.cs
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations;
using System.Net;

namespace api.Data.Models
{
    public class Customer
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Address { get; set; }

        [Required]
        public string Phone { get; set; }

        public ICollection<Order> Orders { get; } = new HashSet<Order>();
    }
}

File: client/src/app/shared/types/customer.ts
import { Order } from "./order";

export interface Customer {
  id: number;
  name: string;
  address: string;
  phone: string;
  orders: Order[];
}

export interface CustomerForResponse {
  customers: Customer[];
};

export type CustomerForRequest = {
  name: string;
  address: string;
  phone: string;
}
